packages/engine/src/avatar/functions/avatarFunctions.ts

```diff
@@ -314,39 +328,45 @@ export function makeSkinnedMeshFromBoneData(bonesData): SkinnedMesh {
   return skinnedMesh
 }
 
+/**
+ * Find the head bone
+ * @param root
+ */
+
+function findHeadBone(root: Object3D) {
+  let headBone = null
+  root.traverse((object) => {
+    if (!(object as any).isSkinnedMesh) return
+    const bones = (object as any).skeleton.bones
+    const sortedBone = bones.find((bone) => /head/i.test(bone.name))
+    if (!sortedBone) return
+    headBone = sortedBone
+  })
+  return headBone
+}
+
 /**
  * Adds required parameters to mesh's material
  * to enable avatar's head decapitation (opacity fade)
  * @param model
  * @param material
  */
-function setupHeadDecap(model: Object3D, material: Material) {
-  const mesh = model.getObjectByProperty('type', 'SkinnedMesh') as SkinnedMesh
-
-  if (!mesh) {
-    console.warn("Could not find object's SkinnedMesh", model)
-    return
-  }
-
-  const bones = mesh.skeleton.bones
-  const headBone = bones.find((bone) => /head/i.test(bone.name))
-
-  if (!headBone) {
-    console.warn("Could not find SkinnedMesh's head bone", mesh)
-    return
-  }
-
+function setupHeadDecap(object: any, headBone: any | undefined, material: Material) {
   // Create a copy of the mesh to hide 'internal' polygons when opacity is below 1
-  const skinnedMeshMask = SkeletonUtils.clone(model).getObjectByProperty('type', 'SkinnedMesh') as SkinnedMesh
-  mesh.parent?.add(skinnedMeshMask)
-  skinnedMeshMask.skeleton = mesh.skeleton
-  skinnedMeshMask.bindMatrix = mesh.bindMatrix
-  skinnedMeshMask.bindMatrixInverse = mesh.bindMatrixInverse
-  skinnedMeshMask.material = new MeshBasicMaterial({ skinning: true, colorWrite: false } as any)
-  skinnedMeshMask.name = skinnedMeshMask.name + '_Mask'
-  skinnedMeshMask.renderOrder = 1
-  ;(mesh.material as any).depthWrite = false
+  if (material.opacity > 0) {
```

Comment:
The material's opacity changes during runtime. The correct way would be to check the opacity value on each update and set skinnedMeshMask.visible value based on that.